### PR TITLE
added mob programming content

### PR DESCRIPTION
**Pull request by phiros**
The purpose of this PR is to add a basic but useful description of pair programming.

- [x] description
- [x] requirements
- [x] resources
  - [x] books
  - [x] web resources
  - [x] videos
- [x] relations
- [x] take your action
  - [x] for programmers new to mob programming
  - [x] next level & things to try out

### CODE REVIEW

**Comment by phiros** [comment]
I didn't add a good video for mob programming since I couldn't find one. I'd prefer to add one later. 
Apart from that, this PR can be reviewed.

File: definitions/content/mob_programming.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+type: content
+tags: practice
```
Comment (smoothd) [inline definitions/content/mob_programming.md]: Please consider a list of tags []

File: definitions/content/mob_programming.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
...for working as a group on software. There are different modes of operation. One of the simplest mode to get started with, requires one computer, a big screen and three roles. The roles are: A driver (singular), a navigator (singular) and a pool of experts. The driver is the typist and transforms ideas communicated by the navigator into code. The driver does not have any ideas of their own and only listens to what the navigator says. The navigator task is to give the driver high level instructions on how to solve the problem at hand without dictating code. Therefore, the navigator thinks strategially about the problem to be solved (e.g. "In order to implement the new REST API we should create an appropriately named controller"). The driver thinks tactically (e.g. "I need to create a controller with the name FooController that inherents from the RestApi class"). The pool of experts is everyone else in the team who isn't a driver or navigator. The pool of experts can be queried by the navigator whenever the navigator feels help is needed (e.g. the navigator feels stuck). They can discuss what is going on among themselves but can only communicate with the navigator when being asked to.
```
Comment (smoothd) [inline definitions/content/mob_programming.md]: Please change _navigator task_ to _navigator's task_

Comment (smoothd) [inline definitions/content/mob_programming.md]: Please consider inserting newlines to create paragraphs.

**Comment by smoothd** [comment]
LGTM

**Pull request merged**